perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_ues_attribute_def_def_isCesnetEligibleLastSeen.java

```diff
@@ -0,0 +1,84 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.User;
+import cz.metacentrum.perun.core.api.UserExtSource;
+import cz.metacentrum.perun.core.api.exceptions.AttributeNotExistsException;
+import cz.metacentrum.perun.core.api.exceptions.ConsistencyErrorException;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.UserNotExistsException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeAssignmentException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeValueException;
+import cz.metacentrum.perun.core.api.exceptions.WrongReferenceAttributeValueException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.UserExtSourceAttributesModuleAbstract;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Timestamp;
+
+/**
+ * Attribute module for isCesnetEligibleLastSeen, value is String representing timestamp.
+ *
+ * @author Radoslav Čerhák <r.cerhak@gmail.com>
+ */
+public class urn_perun_ues_attribute_def_def_isCesnetEligibleLastSeen extends UserExtSourceAttributesModuleAbstract {
+
+	private static final String A_USER_DEF_IS_CESNET_ELIGIBLE_LAST_SEEN = AttributesManager.NS_USER_ATTR_DEF + ":isCesnetEligibleLastSeen";
+	private static final Logger log = LoggerFactory.getLogger(urn_perun_ues_attribute_def_def_isCesnetEligibleLastSeen.class);
+
+	/**
+	 * When isCesnetEligibleLastSeen of a user's ext source is set, check if it is
+	 * more recent than the user's current isCesnetEligibleLastSeen. If so, update it.
+	 *
+	 * @param session
+	 * @param ues
+	 * @param attribute
+	 */
+	@Override
+	public void changedAttributeHook(PerunSessionImpl session, UserExtSource ues, Attribute attribute) {
+		if (attribute.getValue() == null) return;
+
+		User user;
+		try {
+			user = session.getPerunBl().getUsersManagerBl().getUserById(session, ues.getUserId());
+		} catch (UserNotExistsException ex) {
+			throw new ConsistencyErrorException(ex);
+		}
+
+		Attribute userIsCesnetEligible;
+		try {
+			userIsCesnetEligible = session.getPerunBl().getAttributesManagerBl().getAttribute(session, user, AttributesManager.NS_USER_ATTR_DEF + ":isCesnetEligibleLastSeen");
+		} catch (AttributeNotExistsException ex) {
+			log.warn("Attribute {} doesn't exist.", A_USER_DEF_IS_CESNET_ELIGIBLE_LAST_SEEN);
+			return;
+		} catch (WrongAttributeAssignmentException ex) {
+			throw new InternalErrorException(ex);
+		}
+
+		Timestamp uesTimestamp = Timestamp.valueOf(attribute.valueAsString());
```

Comment:
Are we sure that the ues attribute will always contain the correct timestamp? Maybe we should create method checkAttributeSyntax which ensures that there is not some random string. What do you think @zlamalp ?

Reply:
Good point @balcirakpeter, we should have syntax check method and same should be done for user version of attribute.